src/runtime/forester.rs

```diff
@@ -225,17 +230,26 @@ impl Forester {
                                             debug!(target:"flow[run]", "tick:{}, {tpe}. stay with the new state: {}",ctx.curr_ts(),&ns);
                                             ctx.new_state(id, ns)?;
                                         }
+                                        FlowDecision::Halt { .. } => {
+                                            unreachable!("A child returning running should not trigger a flow node halt decision.");
```

Comment:
maybe better to return an error immediately here?
```
return Err(RuntimeError::Unexpected("A child returning running should not trigger a flow node halt decision."))
```

Reply:
Good suggestion, fixed.